ui/packages/catalog-ui-search/src/main/webapp/component/query-advanced/query-advanced.view.js

```diff
@@ -90,12 +95,14 @@ module.exports = Marionette.LayoutView.extend({
     },
     save: function(){
         this.$el.removeClass('is-editing');
+        var inputValue = this.$el.find('intrigue-input > div.if-editing > input').val();
         this.querySettings.currentView.saveToModel();
 
         this.queryAdvanced.currentView.sortCollection();
         this.model.set({
             cql: this.options.isFormBuilder !== true ? this.queryAdvanced.currentView.transformToCql() : "",
-            filterTree: JSON.stringify(this.queryAdvanced.currentView.getFilters())
+            filterTree: JSON.stringify(this.queryAdvanced.currentView.getFilters()),
+            previousAnyText: inputValue
```

Comment:
If we modify the query-add component a bit we can get by without the previousAnyText attribute.  Let me know if you have some time to screenshare / vscode collaborate on this.